### PR TITLE
flow typings for node's built-in `Module` module

### PR DESCRIPTION
**Pull request by gorakong**
Test Plan:
- verified that there were no flow errors

### CODE REVIEW

**Comment by mischnic** [comment]
Maybe we should also try upstreaming this: https://github.com/facebook/flow/blob/05cd1bae10d89c9d4155ab99d2d1a9ed5eca9f9b/lib/node.js#L125-L130

**Comment by gorakong** [comment]
@mischnic sounds good! will be following up with a submission to the Node.js libdef in the main flow repo

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 7.26s | -45.00ms |
| Cached | 1.73s | +48.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 187.00ms | -93.00ms 🚀 |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 262.00ms | +45.00ms ⚠️ |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 1.34s | +80.00ms ⚠️ |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 1.38s | +117.00ms ⚠️ |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 26.70s | -15.00ms |
| Cached | 1.69s | -5.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 140.00ms | +8.00ms ⚠️ |

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 2.68m | -3.26s |
| Cached | 4.84s | +53.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/EmojiPickerComponent.8d912556.js | 139.08kb | +0.00b | 11.19s | -618.00ms 🚀 |
| dist/Modal.cf8447b4.js | 39.02kb | +0.00b | 6.29s | +920.00ms ⚠️ |
| dist/component.42a949b6.js | 30.87kb | +0.00b | 2.99s | +196.00ms ⚠️ |
| dist/esm.74ff7b72.js | 27.74kb | +0.00b | 1.42m | +1.23m ⚠️ |
| dist/component.22b0fde0.js | 22.51kb | +0.00b | 6.29s | +920.00ms ⚠️ |
| dist/DatePicker.329325bd.js | 21.16kb | +0.00b | 1.42m | +1.22m ⚠️ |
| dist/js.0c4ce585.js | 16.55kb | +0.00b | 6.29s | +919.00ms ⚠️ |
| dist/workerHasher.a22a1398.js | 11.90kb | +0.00b | 11.46s | -1.27m 🚀 |
| dist/component.601a47de.js | 6.27kb | +0.00b | 7.50s | -579.00ms 🚀 |
| dist/EmojiPickerComponent.1a428d49.js | 3.68kb | +0.00b | 11.46s | -1.27m 🚀 |
| dist/png-chunks-extract.08e8af36.js | 3.55kb | +0.00b | 6.29s | +920.00ms ⚠️ |
| dist/Modal.fca4b016.js | 3.15kb | +0.00b | 2.99s | +199.00ms ⚠️ |
| dist/clipboard.dfcf0480.js | 2.97kb | +0.00b | 54.09s | -33.42s 🚀 |
| dist/ResourcedEmojiComponent.cc7f53a2.js | 2.15kb | +0.00b | 12.45s | -892.00ms 🚀 |
| dist/date.65bb6f77.js | 1.96kb | +0.00b | 7.05s | +871.00ms ⚠️ |
| dist/images.f4282212.js | 1.90kb | +0.00b | 7.05s | +871.00ms ⚠️ |
| dist/feedback.946661fe.js | 1.86kb | +0.00b | 7.59s | +473.00ms ⚠️ |
| dist/16.6d4b938b.js | 1.86kb | +0.00b | 6.71s | +869.00ms ⚠️ |
| dist/browser.0e955b5c.js | 1.82kb | +0.00b | 1.42m | +32.62s ⚠️ |
| dist/list-number.605d56ab.js | 1.68kb | +0.00b | 7.22s | +904.00ms ⚠️ |
| dist/status.69eb06af.js | 1.68kb | +0.00b | 7.47s | +614.00ms ⚠️ |
| dist/code.3c7bf0dd.js | 1.61kb | +0.00b | 6.87s | +832.00ms ⚠️ |
| dist/link.72eaa0b1.js | 1.53kb | +0.00b | 7.22s | +904.00ms ⚠️ |
| dist/heading6.7a56e9b9.js | 1.53kb | +0.00b | 7.59s | +473.00ms ⚠️ |
| dist/heading3.44db9957.js | 1.51kb | +0.00b | 7.52s | +410.00ms ⚠️ |
| dist/16.2ac6f1f7.js | 1.51kb | +0.00b | 6.52s | +905.00ms ⚠️ |
| dist/16.24210346.js | 1.46kb | +0.00b | 7.56s | -519.00ms 🚀 |
| dist/16.c087bab3.js | 1.46kb | +0.00b | 6.52s | +908.00ms ⚠️ |
| dist/emoji.7fac1ca5.js | 1.45kb | +0.00b | 7.05s | +871.00ms ⚠️ |
| dist/16.72a1f2e3.js | 1.45kb | +0.00b | 3.00s | +199.00ms ⚠️ |
| dist/16.7871a52d.js | 1.44kb | +0.00b | 6.87s | +832.00ms ⚠️ |
| dist/16.1a8644cf.js | 1.40kb | +0.00b | 6.87s | +832.00ms ⚠️ |
| dist/heading5.7de42d71.js | 1.40kb | +0.00b | 7.59s | +473.00ms ⚠️ |
| dist/expand.b1f08bc4.js | 1.38kb | +0.00b | 1.42m | +1.22m ⚠️ |
| dist/16.a1a8e6ae.js | 1.36kb | +0.00b | 6.71s | +869.00ms ⚠️ |
| dist/16.abea8c40.js | 1.34kb | +0.00b | 3.00s | +200.00ms ⚠️ |
| dist/heading2.b89a5450.js | 1.33kb | +0.00b | 7.47s | +614.00ms ⚠️ |
| dist/16.29a7e88c.js | 1.32kb | +0.00b | 6.71s | +869.00ms ⚠️ |
| dist/16.c7ded95d.js | 1.31kb | +0.00b | 6.87s | +832.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.ff849aa7.js | 1.30kb | +0.00b | 31.64s | -18.53s 🚀 |
| dist/mention.51f20c83.js | 1.29kb | +0.00b | 7.22s | +904.00ms ⚠️ |
| dist/heading4.f718f48c.js | 1.29kb | +0.00b | 7.59s | +473.00ms ⚠️ |
| dist/Modal.df7b74de.js | 1.28kb | +0.00b | 6.29s | +920.00ms ⚠️ |
| dist/16.0b71835a.js | 1.27kb | +0.00b | 3.00s | +199.00ms ⚠️ |
| dist/layout.85f4f10e.js | 1.27kb | +0.00b | 7.21s | +900.00ms ⚠️ |
| dist/16.4d524300.js | 1.27kb | +0.00b | 6.71s | +868.00ms ⚠️ |
| dist/16.82077efb.js | 1.26kb | +0.00b | 6.52s | +908.00ms ⚠️ |
| dist/16.d881cb16.js | 1.26kb | +0.00b | 6.52s | +907.00ms ⚠️ |
| dist/16.ae604df2.js | 1.26kb | +0.00b | 6.52s | +905.00ms ⚠️ |
| dist/16.1a3f7446.js | 1.26kb | +0.00b | 7.50s | -581.00ms 🚀 |
| dist/divider.a3b503cc.js | 1.25kb | +0.00b | 7.05s | +871.00ms ⚠️ |
| dist/quote.07ebf058.js | 1.25kb | +0.00b | 7.42s | +569.00ms ⚠️ |
| dist/action.ad8dc9b0.js | 1.23kb | +0.00b | 6.87s | +832.00ms ⚠️ |
| dist/decision.0401e313.js | 1.21kb | +0.00b | 7.05s | +871.00ms ⚠️ |
| dist/panel-warning.736ed649.js | 1.21kb | +0.00b | 7.33s | +765.00ms ⚠️ |
| dist/16.e5321925.js | 1.18kb | +0.00b | 6.71s | +869.00ms ⚠️ |
| dist/list.2f77057e.js | 1.18kb | +0.00b | 7.22s | +904.00ms ⚠️ |
| dist/heading1.edaa8b50.js | 1.18kb | +0.00b | 7.47s | +614.00ms ⚠️ |
| dist/panel-error.a4fdbb96.js | 1.11kb | +0.00b | 7.29s | +719.00ms ⚠️ |
| dist/panel.4c093c14.js | 1.10kb | +0.00b | 7.33s | +765.00ms ⚠️ |
| dist/table.70f6910c.js | 1.09kb | +0.00b | 7.47s | +614.00ms ⚠️ |
| dist/panel-success.d387a630.js | 1.05kb | +0.00b | 7.33s | +765.00ms ⚠️ |
| dist/panel-note.639e3cf0.js | 1.05kb | +0.00b | 7.33s | +765.00ms ⚠️ |
| dist/simpleHasher.7d3e36e5.js | 755.00b | +0.00b | 11.46s | -1.27m 🚀 |
| dist/index.html | 119.00b | +0.00b | 5.68s | +397.00ms ⚠️ |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.3f2dab81.js | 2.30mb | +0.00b | 500.00ms | +75.00ms ⚠️ |
| dist/pdfRenderer.c185dd8c.js | 1.11mb | +0.00b | 746.00ms | +81.00ms ⚠️ |
| dist/editorView.30cf4b8b.js | 587.39kb | +0.00b | 777.00ms | +79.00ms ⚠️ |
| dist/popup.36a19e03.js | 169.21kb | +0.00b | 777.00ms | +98.00ms ⚠️ |
| dist/EmojiPickerComponent.8d912556.js | 139.08kb | +0.00b | 712.00ms | +69.00ms ⚠️ |
| dist/Toolbar.a86f03bd.js | 92.41kb | +0.00b | 777.00ms | +53.00ms ⚠️ |
| dist/media-viewer.02ddb3dd.js | 72.87kb | +0.00b | 746.00ms | +82.00ms ⚠️ |
| dist/card.83cc2b75.js | 55.07kb | +0.00b | 735.00ms | +83.00ms ⚠️ |
| dist/card.89d72bb0.js | 52.83kb | +0.00b | 761.00ms | +72.00ms ⚠️ |
| dist/esm.74ff7b72.js | 27.74kb | +0.00b | 712.00ms | +69.00ms ⚠️ |
| dist/component.22b0fde0.js | 22.51kb | +0.00b | 513.00ms | +52.00ms ⚠️ |
| dist/DatePicker.6f5e2cdb.js | 21.03kb | +0.00b | 685.00ms | +49.00ms ⚠️ |
| dist/dropzone.0ba9c468.js | 17.38kb | +0.00b | 762.00ms | +73.00ms ⚠️ |
| dist/smartMediaEditor.bcfaaa75.js | 16.93kb | +0.00b | 775.00ms | +77.00ms ⚠️ |
| dist/js.0c4ce585.js | 16.55kb | +0.00b | 513.00ms | +65.00ms ⚠️ |
| dist/ui.74b088fb.js | 14.11kb | +0.00b | 735.00ms | +83.00ms ⚠️ |
| dist/workerHasher.a22a1398.js | 11.90kb | +0.00b | 712.00ms | +69.00ms ⚠️ |
| dist/component.601a47de.js | 6.27kb | +0.00b | 518.00ms | +53.00ms ⚠️ |
| dist/card.fbae841f.js | 5.76kb | +0.00b | 746.00ms | +82.00ms ⚠️ |
| dist/media-viewer.09780698.js | 4.01kb | +0.00b | 746.00ms | +81.00ms ⚠️ |
| dist/EmojiPickerComponent.3548646f.js | 3.56kb | +0.00b | 735.00ms | +83.00ms ⚠️ |
| dist/png-chunks-extract.08e8af36.js | 3.55kb | +0.00b | 513.00ms | +65.00ms ⚠️ |
| dist/index.a3f66c21.css | 3.46kb | +0.00b | 816.00ms | +49.00ms ⚠️ |
| dist/dropzone.a85bfb74.js | 3.44kb | +0.00b | 777.00ms | +98.00ms ⚠️ |
| dist/Modal.fca4b016.js | 3.15kb | +0.00b | 504.00ms | +74.00ms ⚠️ |
| dist/clipboard.dfcf0480.js | 2.97kb | +0.00b | 760.00ms | +72.00ms ⚠️ |
| dist/16.29c076d0.js | 2.49kb | +0.00b | 499.00ms | +71.00ms ⚠️ |
| dist/ResourcedEmojiComponent.cc7f53a2.js | 2.15kb | +0.00b | 735.00ms | +83.00ms ⚠️ |
| dist/card.b48be6f2.js | 2.15kb | +0.00b | 777.00ms | +98.00ms ⚠️ |
| dist/feedback.946661fe.js | 1.86kb | +0.00b | 685.00ms | +60.00ms ⚠️ |
| dist/browser.0e955b5c.js | 1.82kb | +0.00b | 760.00ms | +71.00ms ⚠️ |
| dist/16.07e17d98.js | 1.79kb | +0.00b | 527.00ms | +32.00ms ⚠️ |
| dist/workerHasher.0325314f.js | 1.75kb | +0.00b | 712.00ms | +69.00ms ⚠️ |
| dist/workerHasher.01f92ec8.js | 1.75kb | +0.00b | 775.00ms | +77.00ms ⚠️ |
| dist/status.69eb06af.js | 1.68kb | +0.00b | 626.00ms | +60.00ms ⚠️ |
| dist/link.72eaa0b1.js | 1.53kb | +0.00b | 574.00ms | +29.00ms ⚠️ |
| dist/heading6.7a56e9b9.js | 1.53kb | +0.00b | 685.00ms | +59.00ms ⚠️ |
| dist/heading3.44db9957.js | 1.51kb | +0.00b | 651.00ms | +62.00ms ⚠️ |
| dist/16.24210346.js | 1.46kb | +0.00b | 527.00ms | +32.00ms ⚠️ |
| dist/16.c087bab3.js | 1.46kb | +0.00b | 531.00ms | +27.00ms ⚠️ |
| dist/heading5.7de42d71.js | 1.40kb | +0.00b | 685.00ms | +60.00ms ⚠️ |
| dist/expand.b1f08bc4.js | 1.38kb | +0.00b | 685.00ms | +60.00ms ⚠️ |
| dist/heading2.b89a5450.js | 1.33kb | +0.00b | 651.00ms | +62.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.ff849aa7.js | 1.30kb | +0.00b | 777.00ms | +98.00ms ⚠️ |
| dist/heading4.f718f48c.js | 1.29kb | +0.00b | 651.00ms | +59.00ms ⚠️ |
| dist/Modal.df7b74de.js | 1.28kb | +0.00b | 513.00ms | +65.00ms ⚠️ |
| dist/16.82077efb.js | 1.26kb | +0.00b | 527.00ms | +32.00ms ⚠️ |
| dist/16.1a3f7446.js | 1.26kb | +0.00b | 527.00ms | +32.00ms ⚠️ |
| dist/quote.07ebf058.js | 1.25kb | +0.00b | 597.00ms | +36.00ms ⚠️ |
| dist/component.9bdd6688.js | 1.23kb | +0.00b | 499.00ms | +74.00ms ⚠️ |
| dist/16.7ed14d80.js | 1.23kb | +0.00b | 499.00ms | +74.00ms ⚠️ |
| dist/panel-warning.736ed649.js | 1.21kb | +0.00b | 597.00ms | +36.00ms ⚠️ |
| dist/heading1.edaa8b50.js | 1.18kb | +0.00b | 651.00ms | +62.00ms ⚠️ |
| dist/panel.4c093c14.js | 1.10kb | +0.00b | 597.00ms | +36.00ms ⚠️ |
| dist/table.70f6910c.js | 1.09kb | +0.00b | 651.00ms | +62.00ms ⚠️ |
| dist/media-picker-analytics-error-boundary.87a6efed.js | 1.05kb | +0.00b | 777.00ms | +92.00ms ⚠️ |
| dist/panel-success.d387a630.js | 1.05kb | +0.00b | 597.00ms | +36.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.71c67c81.js | 1.05kb | +0.00b | 735.00ms | +77.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.3605963d.js | 1.05kb | +0.00b | 761.00ms | +67.00ms ⚠️ |
| dist/panel-note.639e3cf0.js | 1.05kb | +0.00b | 585.00ms | +30.00ms ⚠️ |
| dist/media-viewer-analytics-error-boundary.cd02e050.js | 995.00b | +0.00b | 746.00ms | +79.00ms ⚠️ |
| dist/simpleHasher.7d3e36e5.js | 755.00b | +0.00b | 712.00ms | +66.00ms ⚠️ |
| dist/simpleHasher.a0d51de0.js | 755.00b | +0.00b | 775.00ms | +77.00ms ⚠️ |
| dist/index.html | 119.00b | +0.00b | 294.00ms | +35.00ms ⚠️ |

</p></details>

<details><summary>Three.js x4 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/282576933956354573)**

File: flow-libs/module.js.flow [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+// Derived from the documentation and source of Module located at
+// https://nodejs.org/api/module.html#module_modules_module_api and
+// https://github.com/nodejs/node/blob/cbfa2d11d4e42ea7f6a847a6930fd1022773252e/lib/internal/modules/cjs/loader.js/
+// Which is licensed MIT
+
+declare module 'module' {
+  declare type FilePath = string;
+
+  // Derived from https://github.com/facebook/flow/blob/61913b9f0dcd564fd4c74fefbe896b33ef447cab/lib/node.js
+  declare class Url {
```
Comment (wbinnssmith) [inline flow-libs/module.js.flow:12]: `import type {URL} from 'url'`?

Comment (wbinnssmith) [reply flow-libs/module.js.flow:12]: looks like this is addressed in the upstream PR :) https://github.com/facebook/flow/pull/8540

**Pull request closed without merging**